chain/blockchain.go

```diff
@@ -72,6 +73,24 @@ func (bc *Blockchain) GetHeader(hash Uint256) (*block.Header, error) {
 }
 
 func (bc *Blockchain) SaveBlock(block *block.Block, fastAdd bool) error {
+	if !fastAdd {
+		if !block.IsHeaderChecked {
+			err := HeaderCheck(block.Header)
+			if err != nil {
+				return err
+			}
+			block.IsHeaderChecked = true
+		}
+
+		if !block.IsTransactionsChecked {
+			err := TransactionCheck(context.Background(), block)
+			if err != nil {
+				return err
+			}
+			block.IsTransactionsChecked = true
```

Comment:
Same as above. Short name for srcCode concision.

Reply:
done